egs/tedlium/s5/local/chain/run_tdnn.sh

```diff
@@ -0,0 +1,205 @@
+## %WER 13.8 | 1155 27512 | 89.4 7.5 3.1 3.2 13.8 87.9 | 0.101 | exp/chain/tdnn/decode_test/score_10_0.0/ctm.filt.filt.sys
+## %WER 12.9 | 1155 27512 | 90.1 6.6 3.3 2.9 12.9 86.1 | 0.043 | exp/chain/tdnn/decode_test_rescore/score_10_0.0/ctm.filt.filt.sys
+# The final WER (rescored WER on the test set) is what we are interested in.
+
+# To reproduce the setup used in the paper, set the following variables:
+# affix=_more_ce
+# relu_dim=525
+# xent_regularize=0.2
+#
+# Results: (Run for x in exp/chain/tdnn_more_ce_sp/decode*; do [ -d $x ] && grep Sum $x/score_*/*.sys | utils/best_wer.sh; done 2>/dev/null)
+## %WER 14.3 | 507 17792 | 89.0 7.8 3.2 3.3 14.3 93.5 | 0.116 | exp/chain/tdnn_more_ce_sp/decode_dev/score_10_0.0/ctm.filt.filt.sys
+## %WER 13.0 | 507 17792 | 90.0 6.9 3.2 2.9 13.0 91.3 | -0.003 | exp/chain/tdnn_more_ce_sp/decode_devv_rescore/score_10_0.0/ctm.filt.filt.sys
+## %WER 13.8 | 1155 27512 | 89.1 7.4 3.4 2.9 13.8 87.5 | 0.082 | exp/chain/tdnn_more_ce_sp/decode_test/score_10_0.5/ctm.filt.filt.sys
+## %WER 12.8 | 1155 27512 | 90.4 6.6 3.1 3.1 12.8 86.7 | 0.014 | exp/chain/tdnn_more_ce_sp/decode_test_rescore/score_10_0.0/ctm.filt.filt.sys
+
+set -uo pipefail
+
+# configs for 'chain'
+affix=
+stage=0 # After running the entire script once, you can set stage=12 to tune the neural net only.
+train_stage=-10
+get_egs_stage=-10
+dir=exp/chain/tdnn
+decode_iter=
+
+# TDNN options
+# this script uses the new tdnn config generator so it needs a final 0 to reflect that the final layer input has no splicing
+self_repair_scale=0.00001
+# training options
+num_epochs=4
+initial_effective_lrate=0.001
+final_effective_lrate=0.0001
+leftmost_questions_truncate=-1
+max_param_change=2.0
+final_layer_normalize_target=0.5
+num_jobs_initial=3
+num_jobs_final=8
+minibatch_size=128
+relu_dim=425
+frames_per_eg=150
+remove_egs=false
+xent_regularize=0.1
+
+
+# End configuration section.
+echo "$0 $@"  # Print the command line for logging
+
+. ./cmd.sh
+. ./path.sh
+. ./utils/parse_options.sh
+
+dir=${dir}${affix}
+
+if ! cuda-compiled; then
+  cat <<EOF && exit 1
+This script is intended to be used with GPUs but you have not compiled Kaldi with CUDA
+If you want to use GPUs (and have them), go to src/, and configure and make on a machine
+where "nvcc" is installed.
+EOF
+fi
+
+# The iVector-extraction and feature-dumping parts are the same as the standard
+# nnet3 setup, and you can skip them by setting "--stage 9" if you have already
+# run those things.
+
+train_set=train
+gmm_dir=exp/tri3
+ali_dir=exp/tri3_ali
+lats_dir=${ali_dir/ali/lats}
+treedir=exp/chain/tri3_tree
+lang=data/lang_chain
+
+mkdir -p $dir
+# It is not required to generate alignments here, but doing so
+# allows us to run local/nnet3 models without having to run this script first.
+# If you don't want to run those models, you can do "--generate-alignments false"
+local/nnet3/run_ivector_common.sh --stage $stage \
+  --generate-alignments true \
```

Comment:
we usually run just the best model so use `--generate-alignments false` as the default.